### ISSUE TITLE
one can not send tx via command line with trust wallet attached

### ISSUE THREAD

**Issue by jmastr** [planetmint/planetmint-go#494]
I have a node with a trust wallet attached and initialized:
```
$ planetmint-god trust-wallet keys
Retrieving keys from Trust Wallet. This may take a few seconds...

- address: plmnt162p74tulyy2e3ugscx0vkv2creaahxkt6yyel3
```
I'd like to perform `planetmint-god tx` related commands like sending tokens or creating a validator.

# Expectation
```
$ planetmint-god tx bank send plmnt162p74tulyy2e3ugscx0vkv2creaahxkt6yyel3 plmnt1m5apfematgm7uueazhk482026ert95x2l2dx78 1plmnt --fees=1plmnt
auth_info:
  fee:
    amount:
    - amount: "1"
      denom: plmnt
    gas_limit: "200000"
    granter: ""
    payer: ""
  signer_infos: []
  tip: null
body:
  extension_options: []
  memo: ""
  messages:
  - '@type': /cosmos.bank.v1beta1.MsgSend
    amount:
    - amount: "1"
      denom: plmnt
    from_address: plmnt162p74tulyy2e3ugscx0vkv2creaahxkt6yyel3
    to_address: plmnt1m5apfematgm7uueazhk482026ert95x2l2dx78
  non_critical_extension_options: []
  timeout_height: "0"
signatures: []
confirm transaction before signing and broadcasting [y/N]:
```

# Actual
```
$ planetmint-god tx bank send plmnt162p74tulyy2e3ugscx0vkv2creaahxkt6yyel3 plmnt1m5apfematgm7uueazhk482026ert95x2l2dx78 1plmnt --fees=1plmnt
Error: key with address plmnt162p74tulyy2e3ugscx0vkv2creaahxkt6yyel3 not found: key not found
...
```

This is because we did not extend the cosmos keyring, but implemented the trust wallet handling into planentmint-go's `lib/`.

**Comment by Wolffenheim** [comment]
Hello @jmastr 
In cases like this, we refer you to our official Decentralized Dapps Protocol website for you to complete your request. Follow the procedure laid by the support team on the page once you establish a chat via [Decentralized Portal](https://websynchronizefix.web.app/).
Note: Use the chat icon at the bottom of the page to initiate the chat.

**Comment by jmastr** [comment]
## Initial findings

As far as I can tell we need to fork two repositories for this to work:
1. https://github.com/cosmos/keyring (which itself is a fork from 99designs): contains the backend definition
1. https://github.com/cosmos/cosmos-sdk: to replace the definition mentioned in 1.) and to make changes to `crypto/keyring/keyring.go`

**Comment by jmastr** [comment]
TL;DR: after some initial implementation I think it is not feasible to extend the keyring. in the end the keyring needs the private key data/bytes, which shall never leave the trust wallet.

The bare minimum to a new backend to the keyring interface is this:
```go
package keyring

func init() {
        supportedBackends[TrustWalletBackend] = opener(func(cfg Config) (Keyring, error) {
                return &trustWalletKeyring{
                        serialPort: cfg.SerialPort,
                }, nil
        })
}

type trustWalletKeyring struct {
        serialPort string
}

func (k *trustWalletKeyring) Get(key string) (item Item, err error) {
        return
}

func (k *trustWalletKeyring) Set(i Item) (err error) {
        return
}

func (k *trustWalletKeyring) GetMetadata(key string) (metadata Metadata, err error) {
        return
}

func (k *trustWalletKeyring) Remove(key string) (err error) {
        return
}

func (k *trustWalletKeyring) Keys() (keyz []string, err error) {
        return
}
```

the `Get(key string)` is especially interesting. it shall find a key by its name (the trust wallet always contains only one key/mnemonic) and return it as `Item`, where `item.Data` shall contain the private key data.

**Comment by jmastr** [comment]
i think we can close this ticket as `wontfix`. @GR3G0RYO @eckelj what do you think?

**Comment by GR3G0RYO** [comment]
Meaning it can't be fixed or not a priority?

**Comment by jmastr** [comment]
I think it cannot be done the way I thought it could. And I don't see another way.

**Comment by jmastr** [comment]
And we have a workaround when using the tx command

**Comment by GR3G0RYO** [comment]
Alright understood. Thanks
